test_tools/healthcheck_test_tool/main.go

```diff
@@ -97,51 +101,54 @@ func doHTTPCheck(target net.IP, secure bool) {
 	hc.Method = *method
 	hc.Proxy = *proxy
 	hc.TLSVerify = *tlsVerify
-	check(hc)
+	return check(hc)
 }
 
-func doPingCheck(target net.IP) {
+func doPingCheck(target net.IP) error {
+	var err error
 	pc := healthcheck.NewPingChecker(target)
 	pc.Mark = *mark
 	received := 0
 	for i := 0; i < *count; i++ {
 		r := pc.Check(time.Duration(0))
 		if !r.Success {
+			err = fmt.Errorf("Failed to ping %v: %v", target, r)
```

Comment:
instead of returning this error, why not returning the overall summary in the error like on line 123?

Reply:
good suggestion.